src/main/java/org/opensrp/repository/postgres/mapper/custom/CustomStockMetadataMapper.java

```diff
@@ -29,4 +29,8 @@
 	List<PgStockAndProductDetails> selectManyStockAndProductDetailsByServicePointId(@Param("locations") List<String> locations);
 
 	List<PgStockAndProductDetails> selectStockAndProductDetailsByStockId(@Param("documentId") String documentId);
+
+	List<PgStockAndProductDetails> selectManyStockAndProductDetailsBySearchBean(@Param("searchBean") StockSearchBean searchBean,
+			@Param("todayDate") Date todayDate, @Param("orderByClause") String orderByClause, @Param("offset") int offset, @Param("limit") int limit);
+
```

Comment:
seems like we are passing the same params as `selectManyBySearchBean` why not update `selectManyBySearchBean` and add logic where additional join and filters are added depending on `isReturnProduct`

Reply:
`selectManyBySearchBean` returns a list of Stock while here we need a list of `StockAndProductDetails` since it's a collection of Stock and Product. The reason for creating a new method was due to the difference in the result map.

Reply:
Okay then move the where block on `selectManyBySearchBean` to an sql block that can be shared by both methods. This makes it easier to maintain in the futiure

Reply:
Done